src/app/branded/branded-checkout.component.js

```diff
@@ -17,19 +19,25 @@ let componentName = 'brandedCheckout';
 class BrandedCheckoutController {
 
   /* @ngInject */
-  constructor($window, analyticsFactory, tsysService) {
+  constructor($window, analyticsFactory, tsysService, sessionService) {
     this.$window = $window;
     this.analyticsFactory = analyticsFactory;
     this.tsysService = tsysService;
+    this.sessionService = sessionService;
   }
 
   $onInit() {
     this.code = this.designationNumber;
     this.tsysService.setDevice(this.tsysDevice);
-    this.checkoutStep = 'giftContactPayment';
-    this.formatDonorDetails();
     this.analyticsFactory.pageLoaded(true);
-    this.fireAnalyticsEvents('contact', 'payment');
+
+    this.sessionService.signOut().subscribe(() => {
+      this.sessionService.downgradeToGuest(true, false).subscribe(() => {
```

Comment:
I think this just needs to be any, single, request. I talked with Brian about it a while ago and I think that's what we figured out. We need to request a token once and store it in cookies before sending more requests that load the token from cookies otherwise we will keep getting new tokens and storing them and requests with old ones won't work.

You may not need the `verifyRole` flag in the session service if you don't need to call `this.sessionService.downgradeToGuest`.

Is any request sent in parallel with `this.cartService.get()` in step 1? Doesn't look like it. But the sub components may start sending requests. That may have been why I had the big wrapper `ng-if` in `branded-checkout-step-1.tpl.html` to keep sub components from rendering until `this.cartService.get()` had completed.